include/kf/EarlyAllocator.h

```diff
@@ -24,42 +24,44 @@ namespace kf
 
         template<class Y>
         constexpr EarlyAllocator(const EarlyAllocator<Y>& other) noexcept : m_ptr(reinterpret_cast<T*>(other.m_ptr)), m_size(other.m_size)
-        {
-        }
+        { }
 
-        _CONSTEXPR20_DYNALLOC ~EarlyAllocator() = default;
-        _CONSTEXPR20_DYNALLOC EarlyAllocator& operator=(const EarlyAllocator&) = default;
+        ~EarlyAllocator() = default;
+        EarlyAllocator& operator=(const EarlyAllocator&) = default;
 
         template<POOL_TYPE poolType>
         T* initialize(const size_t count)
         {
-            assert(!m_ptr);
-            assert(!m_size);
+            UNREFERENCED_PARAMETER(count);
+
+            ASSERT(!m_ptr);
+            ASSERT(!m_size);
 
             m_size = count * sizeof(T);
             m_ptr = static_cast<T*>(operator new(m_size, poolType));
 
             return m_ptr;
         }
 
-        _CONSTEXPR20_DYNALLOC void deallocate(const T* ptr, const size_t count)
+        void deallocate(const T* ptr, const size_t count)
         {
-            if (ptr != m_ptr || count * sizeof(T) > m_size)
-            {
-                _Xinvalid_argument("ptr != m_ptr || count * sizeof(T) > m_size");
-            }
+            UNREFERENCED_PARAMETER(ptr);
+            UNREFERENCED_PARAMETER(count);
+
+            ASSERTMSG("Wrong pointer", ptr == m_ptr);
+            ASSERTMSG("Wrong count", count * sizeof(T) <= m_size);
 
             operator delete(m_ptr);
             m_ptr = nullptr;
             m_size = 0;
         }
 
-        _NODISCARD _CONSTEXPR20_DYNALLOC T* allocate(const size_t count)
```

Comment:
Let's use `_CONSTEXPR20` instead of `_CONSTEXPR20_DYNALLOC`. Here and in other places. Or just a plain `noexcept`.

Reply:
noexcept looks the best to me

Reply:
Also, [[nodiscard]] attribute looks better than _NODISCARD macro, change it too?